File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/order/jobchain/UnmodifiableJobchain.java
package com.sos.scheduler.engine.kernel.order.jobchain;

import com.sos.scheduler.engine.data.folder.JobChainPath;
import com.sos.scheduler.engine.data.order.OrderId;
import com.sos.scheduler.engine.kernel.job.Job;
import com.sos.scheduler.engine.kernel.order.UnmodifiableOrder;

public interface UnmodifiableJobchain {
    String getName();
    JobChainPath getPath();
    boolean refersToJob(Job o);
    UnmodifiableOrder order(OrderId id);
}

File: engine-kernel/src/main/scala/com/sos/scheduler/engine/kernel/order/jobchain/UnmodifiableJobChain.scala
package com.sos.scheduler.engine.kernel.order.jobchain

import com.sos.scheduler.engine.kernel.job.Job

trait UnmodifiableJobChain {

  //def name: String

  //def path: JobChainPath

  private[order] def refersToJob(o: Job): Boolean
}
